src/lib.rs

```diff
@@ -1484,6 +1487,25 @@ pub fn shallow_scan_for_device(
         .find(|&device| device.vendor_id == vendor_id && device.device_id == device_id)
 }
 
+// same as shallow_scan_for_device but scans all instances, some devices like
+// RoT have multiple instances within each chiplet, we should find and return all.
+pub fn scan_for_all_device_instances(
+    bus: u8,
+    ops: &PciEcamCfgOps,
+    vendor_id: u16,
+    device_id: u16,
+) -> Vec<Device> {
```

Comment:
Can we return a DeviceIterator by applying filter operation on lines 1501-1503 instead of pushing to a Vec? This library does not have any dependency on alloc crate and was wondering if we could keep it that way?

Reply:
Done